notebooks/allen_smFISH.py

```diff
@@ -0,0 +1,187 @@
+# EPY: END markdown
+
+# EPY: START code
+# filtered_backup = deepcopy(s)
+# EPY: END code
+
+# EPY: START markdown
+# If you ever want to visualize the image in the notebook, we've added a widget to do that. The first parameter is an indices dict that specifies which hybridization round, channel, z-slice you want to view. The result is a pageable visualization across that arbitrary set of slices. Below I'm visualizing the first channel, which your codebook tells me is Nmnt. 
+# 
+# [N.B. once you click on the slider, you can page with the arrow keys on the keyboard.]
+# EPY: END markdown
+
+# EPY: START code
+s.image.show_stack({Indices.CH: 0});
+# EPY: END code
+
+# EPY: START code
+from starfish.pipeline.filter.bandpass import Bandpass
+s_bandpass = Bandpass(lshort=0.5, llong=7, threshold=None, truncate=4, verbose=True)
+s_bandpass.filter(s)
+# EPY: END code
+
+# EPY: START markdown
+# For bandpass, there's a point where things get weird, at `c == 0; z <= 14`. In that range the images look mostly like noise. However, _above_ that, they look great + background subtracted! The later stages of the pipeline appear robust to this, though, as no spots are called for the noisy sections. 
+# EPY: END markdown
+
+# EPY: START code
+# I wasn't sure if this clipping was supposed to be by volume or tile. I've done tile here, but it can be easily
+# switched to volume. 
+s_clip = Clip(p_min=10, p_max=100, is_volume=False, verbose=True)
+s_clip.filter(s)
+# EPY: END code
+
+# EPY: START code
+from starfish.pipeline.filter.gaussian_low_pass import GaussianLowPass
+
+sigma=(1, 0, 0)  # filter only in z, do nothing in x, y
+glp = GaussianLowPass(sigma=sigma, is_volume=True, verbose=True)
+glp.filter(s)
+# EPY: END code
+
+# EPY: START markdown
+# Below, because spot finding is so slow when single-plex, we'll pilot this on a max projection to show that the parameters work. Here's what trackpy.locate, which we wrap, produces for a z-projection of channel 1. To do use our plotting methods on z-projections we have to expose some of the starfish internals, which will be improved upon. 
+# EPY: END markdown
+
+# EPY: START code
+from showit import image
+from trackpy import locate
+
+# grab a section from the tensor. 
+ch1 = s.max_proj((Indices.Z))[0, 1]
+
+results = locate(ch1, diameter=3, minmass=250, maxsize=3, separation=5, preprocess=False, percentile=10) 
+results.columns = ['y', 'x', 'intensity', 'r', 'eccentricity', 'signal', 'raw_mass', 'ep']
+# EPY: END code
+
+# EPY: START code
+# plot the z-projection
+image(ch1, size=20, clim=(15, 52))
+
+# draw called spots on top as red circles
+# scale radius plots the red circle at scale_radius * spot radius
+s.image._show_spots(results, ax=plt.gca(), scale_radius=7)
+# EPY: END code
+
+# EPY: START markdown
+# Below spot finding is on the _volumes_ for each channel. This will take about `11m30s`
```

Comment:
if this takes 11m30s, running this in travis will likely trigger timeouts unless the verbose=True triggers some output frequently enough.  may be worth trying out.

Reply:
Got it, you mean the on-master runs, right? Sounds like we should knock the stack size down for testing.